drivers/gpio/gpio_pca95xx.c

```diff
@@ -448,12 +484,149 @@ static int gpio_pca95xx_port_toggle_bits(const struct device *dev,
 	return ret;
 }
 
+static void gpio_pca95xx_interrupt_worker(struct k_work *work)
+{
+	struct gpio_pca95xx_drv_data * const drv_data = CONTAINER_OF(
+		work, struct gpio_pca95xx_drv_data, interrupt_worker);
+	uint16_t input_status, input_cache;
+	uint32_t enabled_int = 0;
+	int ret;
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	input_cache = drv_data->reg_cache.input;
+
+	ret = update_input_regs(drv_data->instance, &input_status);
+	if (ret == 0) {
+		/* Note: PCA Interrupt status is cleared by reading inputs */
+
+		enabled_int = ((input_cache ^ input_status) &
```

Comment:
Won't this signal an interrupt on every detected state change, even if the pin is configured for interrupts on only one edge?

Reply:
yes it would. Fixed